aips/aip-x.md

```diff
@@ -0,0 +1,237 @@
+    )
+}
+```
+
+The master signer can use `authorize`  to grant permissions to a permissioned signer. The smart contracts would use the `check_` APIs to make sure if a permissioned signer is provided, it needs to have the right permission granted by the master signer to perform the corresponding operation.
+
+In the current design, the permission storage is a `CopyableAny -> u256`  mapping stored in the global storage. Smart contract can define the key themselves to create proper domain separation and act as witness pattern.
+
+### Revoking Permission
+
+The master signer should be able to revoke the permission granted to a permissioned signer at any time. The API would look like the following:
+
+```jsx
+module aptos_framework::permissioned_signer {
+    public fun revoke_permission_handle(master: &signer, permission_address: address);
+}
+```
+
+The `signer_from_permissioned` would abort on revoked signer.
+
+### Example: Fungible Asset Transfer Limit
+```jsx
+module aptos_framework::fungible_asset {
+    struct WithdrawPermission has store {
+        token_addr: address,
+    }
+
+    public fun withdraw<T: key>(owner: &signer, store: Object<T>, amount: u64): FungibleAsset {
+        assert!(
+            permissioned_signer::check_permission(
+                owner,
+                amount as u256,
+                WithdrawPermission {
+                    metadata_address: object::object_address(&borrow_store_resource(&store).metadata)
+                }
+            ),
+            error::permission_denied(EWITHDRAW_PERMISSION_DENIED)
+        );
+        //… regular withdraw logic
+    }
+    
+    public fun grant_permission(
+        master: &signer,
+        permissioned: &signer,
+        token_type: Object<Metadata>,
+        amount: u64
+    ) {
+        permissioned_signer::authorize(
+            master,
+            permissioned,
+            amount as u256,
+            WithdrawPermission { metadata_address: object::object_address(&token_type)}
+        )
+    }
+    /// Removing permissions from permissioned signer.
+    public fun revoke_permission(permissioned: &signer, token_type: Object<Metadata>) {
+        permissioned_signer::revoke_permission(permissioned, WithdrawPermission {
+            metadata_address: object::object_address(&token_type),
+        })
+    }
+
+}
+
+```
+
+## Reference Implementation
+
+We have a stacked PRs in https://github.com/aptos-labs/aptos-core/pull/14605. In the stack, we have:
+- Implemented the core library that generates and manages the permissioned signer.
+    - including Move and Rust components.
+- Sweeped throught Aptos Framework codebase with following permissions implemented:
+    - how much fungible asset/coin can be withdrawed from each permissioned signer.
+    - transfer Object.
+    - manipulate NFT object/collection.
+    - operations on account.
+    - allowed to vote
+    - allowed to stake
+    - Privileged operation where master signer is required.
```

Comment:
perhaps you could expand the AIP with snippets on how you handle all these operations? we should have the conversation on why these are the right way to do things by comparison to alternatives.